string start
string stop
string end
---
string[] path